index.js

```diff
@@ -15,13 +15,18 @@ function getSize(el) {
 function useComponentSize(ref) {
   let [ComponentSize, setComponentSize] = useState(getSize(ref.current))
 
-  function handleResize() {
-    if (ref && ref.current) {
-      setComponentSize(getSize(ref.current))
-    }
-  }
+  const handleResize = useCallback(
+    function handleResize() {
+      if (ref.current) {
+        setComponentSize(getSize(ref.current))
+      }
+    },
+    [ref]
+  )
 
   useLayoutEffect(() => {
+    if (!ref.current) { return }
+
     handleResize()
 
     if (ResizeObserver) {
```

Comment:
The `resizeObserver` itself can probably be memoized by using a pair of `useMemo` and a separate `useLayoutEffect`